server/src/test/java/io/crate/integrationtests/TransportSQLActionClassLifecycleTest.java

```diff
@@ -600,7 +600,7 @@ public void selectCurrentTimestamp() throws Exception {
         long before = System.currentTimeMillis();
         SQLResponse response = execute("select current_timestamp(3) from sys.cluster");
         long after = System.currentTimeMillis();
-        assertThat(response.cols()).containsExactly("current_timestamp(3)");
+        assertThat(response.cols()).containsExactly("\"current_timestamp\"(3)");
```

Comment:
>maybe a second method alongside displayName() for the internal purposes

this sounds like a safer option.

Reply:
> the displayName() is used also for the col name in the response.

what is the issue with this?
Adding another method will increase complexity and also surface for issues, so I would only do that when there is a really strong argument for it.

Reply:
I think using `displayName` is indeed the right choice, it's how a identifier should be rendered. Why should we treat this different for expression strings we store in the metadata? I think the opposite, we should ensure that this representation is always the same compared to how we render it to the client/user.

Reply:
This is how it looks like after this fix:
```
cr> select current_timestamp(3);
+------------------------+
| "current_timestamp"(3) |
+------------------------+
|          1724940894987 |
+------------------------+
SELECT 1 row in set (73.626 sec)
cr> select current_timestamp;
+-------------------+
| CURRENT_TIMESTAMP |
+-------------------+
|     1724940943323 |
+-------------------+
SELECT 1 row in set (1.457 sec)
cr> select current_timestamp();
+-------------------+
| CURRENT_TIMESTAMP |
+-------------------+
|     1724940952560 |
+-------------------+
SELECT 1 row in set (1.018 sec)
```
for a column which is mixed case though:
```
cr> create table t("cAmELCaSe" int);
CREATE OK, 1 row affected (0.405 sec)
cr> insert into t values(1);
INSERT OK, 1 row affected (0.030 sec)
cr> select * from t;
+-----------+
| cAmELCaSe |
+-----------+
|         1 |
+-----------+
SELECT 1 row in set (0.015 sec)
```

Reply:
I think then the only open question is whether we treat this as a breaking change?

Reply:
Yeah but isn't the issue different? Isn't the issue that we quote here reserved keywords while they are used at the right place? Like we quote function names which must not be quoted, these names must only be quoted if used somewhere else than at the place they are reserved for.

Reply:
To clarify a bit better: we store mixed case column names without the wrapping `"`.
The reason we need to save the functions like this is that when used in generated columns, we pass the expression through the `SqlParser.createExpression()`, so in order to prevent it from lowercase everything we need to have saved them with the quotes (when needed of course).

I digged a bit more, and actually we cannot just introduce another method similar to displayName, we would need to introduce a new `Style` for the `toString(Style)` method, which is used only for the response col names, which indeed increases complexity.

And imho, if we agree to "break", we should add a breaking change entry.

Reply:
> So we may use a different quoting logic here which won't quote built-in functions if they match reserved keywords?

I was thinking of it but the issue is in the context of `io.crate.expression.symbol.Function` which is unaware of udfs/built-in etc. so I'm not sure if we are able to make the distiction, or you have some idea?

Reply:
What about utilizing `FunctionName.isBuiltin()`?

Reply:
Of course, it was right in front of my :eyes:, big :man_facepalming: 
I had the feeling that `schema == null` would cause other issues, but we cannot create a UDF with null schema, (uses the schema from search_path).
Pushed the commit, please check again.